photutils/psf/daogroup.py

```diff
@@ -0,0 +1,129 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions for implementing the DAOGROUP algorithm proposed by Stetson 1987.
+"""
+
+from __future__ import division
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+__all__ = ['daogroup']
+
+def daogroup(starlist, crit_separation):
+    """
+    This is an implementation of the DAOGROUP algorithm presented by
+    Stetson (1987).
+
+    DAOGROUP divides an entire starlist into sets of distinct, self-contained
+    groups of mutually overlapping stars.
+
+    GROUP accepts as input a list of stars and determines which stars are
+    close enough to be capable of adversely influencing each others' profile
+    fits.
+
+    Parameters
+    ----------
+    starlist : `~astropy.table.Table` or array-like
+        List of stars positions.
+        If `~astropy.table.Table`, columns should be named 'x_0' and 'y_0'.
+        TODO: If array-like, it should be either (x_0, y_0).
+    crit_separation : float or int
+        Distance, in units of pixels, such that any two stars separated by
+        less than this distance will be placed in the same group.
+
+    Returns
+    -------
+    group_starlist : list of `~astropy.table.Table`
+        Each `~astropy.table.Table` in the list corresponds to a group of
+        mutually overlapping starts.
+
+    Notes
+    -----
+    Assuming the psf fwhm to be known, 'crit_separation' may be set to
+    k*fwhm, for some positive real k.
+
+    See
+    ---
+    `~daofind`
+    """
+
+    if not isinstance(crit_separation, (float, int)):
+        raise ValueError('crit_separation is expected to be either float or' +
+                         ' int. Received {}.'.format(type(crit_separation)))
+    if crit_separation < 0.0:
```

Comment:
OK, thinking about it:  = 0 will put each star in its own group. OK, we can do that. Could do it faster by just adding te id column to the table (or splitting it in a group of tables by hand), but we can leave it here for consistency.